staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -453,6 +453,42 @@ type DeleteOptions struct {
 	// foreground.
 	// +optional
 	PropagationPolicy *DeletionPropagation `json:"propagationPolicy,omitempty" protobuf:"varint,4,opt,name=propagationPolicy"`
+
+	// Whether the request should be completed and modifications will not be persisted.
```

Comment:
I'd suggest something more like this:

> When present, indicates that modifications should not be persisted.
> An invalid or unrecognized dryRun directive will result in a BadRequest response and no further processing of the request.

and then we can add details about what directives are supported as we determine that